kroxylicious-kms-providers/kroxylicious-kms-provider-azure-key-vault-kms/src/main/java/io/kroxylicious/kms/provider/azure/config/AzureKeyVaultConfig.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.azure.config;
+
+import java.net.URI;
+import java.util.Objects;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+
+import io.kroxylicious.kms.provider.azure.config.auth.EntraIdentityConfig;
+import io.kroxylicious.proxy.config.tls.Tls;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+@JsonPropertyOrder({ "entraIdentity", "keyVaultBaseUrl", "tls" })
+public record AzureKeyVaultConfig(@JsonProperty(required = true) EntraIdentityConfig entraIdentity,
+                                  @JsonProperty(required = true) String keyVaultBaseUrl,
```

Comment:
Jackson knows about URI, and we've used it on Configuration objects pretty extensively already.